src/app/fdctl/run/tiles/fd_repair.c

```diff
@@ -0,0 +1,675 @@
+#define MAP_NAME     fd_contact_info_table
+#define MAP_KEY_T    fd_pubkey_t
+#define MAP_KEY_EQ   fd_pubkey_eq
+#define MAP_KEY_HASH fd_pubkey_hash
+#define MAP_KEY_COPY fd_pubkey_copy
+#define MAP_T        fd_contact_info_elem_t
+#include "../../../../util/tmpl/fd_map_giant.c"
+
+
+struct fd_repair_tile_ctx {
+  fd_repair_t * repair;
+  fd_repair_config_t repair_config;
+
+  fd_repair_peer_addr_t repair_my_intake_addr;
+  fd_repair_peer_addr_t repair_my_serve_addr;
+  ushort                repair_intake_listen_port;
+  ushort                repair_serve_listen_port;
+
+  uchar       identity_private_key[ 32 ];
+  fd_pubkey_t identity_public_key;
+
+  fd_wksp_t * wksp;
+
+  fd_wksp_t * contact_in_mem;
+  ulong       contact_in_chunk0;
+  ulong       contact_in_wmark;
+
+  fd_wksp_t * stake_weights_in_mem;
+  ulong       stake_weights_in_chunk0;
+  ulong       stake_weights_in_wmark;
+
+  fd_wksp_t * repair_req_in_mem;
+  ulong       repair_req_in_chunk0;
+  ulong       repair_req_in_wmark;
+
+  fd_wksp_t *     net_in;
+  ulong           chunk;
+  ulong           wmark;
+
+  fd_frag_meta_t * net_out_mcache;
+  ulong *          net_out_sync;
+  ulong            net_out_depth;
+  ulong            net_out_seq;
+
+  fd_wksp_t * net_out_mem;
+  ulong       net_out_chunk0;
+  ulong       net_out_wmark;
+  ulong       net_out_chunk;
+
+  fd_frag_meta_t * store_out_mcache;
+  ulong *          store_out_sync;
+  ulong            store_out_depth;
+  ulong            store_out_seq;
+
+  fd_wksp_t * store_out_mem;
+  ulong       store_out_chunk0;
+  ulong       store_out_wmark;
+  ulong       store_out_chunk;
+
+  uchar src_mac_addr[6];
+
+  /* Includes Ethernet, IP, UDP headers */
+  ulong repair_buffer_sz;
+  uchar repair_buffer[ FD_NET_MTU ];
+};
+typedef struct fd_repair_tile_ctx fd_repair_tile_ctx_t;
+
+static fd_gossip_peer_addr_t *
+resolve_hostport( const char * str /* host:port */, fd_gossip_peer_addr_t * res ) {
+  fd_memset( res, 0, sizeof( fd_gossip_peer_addr_t ) );
+
+  /* Find the : and copy out the host */
+  char buf[128];
+  uint i;
+  for( i = 0;; ++i ) {
+    if( str[i] == '\0' || i > sizeof( buf ) - 1U ) {
+      FD_LOG_ERR( ( "missing colon" ) );
+      return NULL;
+    }
+    if( str[i] == ':' ) {
+      buf[i] = '\0';
+      break;
+    }
+    buf[i] = str[i];
+  }
+  if( i == 0 ) /* :port means $HOST:port */
+    gethostname( buf, sizeof( buf ) );
+
+  struct hostent * host = gethostbyname( buf );
+  if( host == NULL ) {
+    FD_LOG_WARNING( ( "unable to resolve host %s", buf ) );
+    return NULL;
+  }
+  /* Convert result to repair address */
+  res->l    = 0;
+  res->addr = ( (struct in_addr *)host->h_addr )->s_addr;
+  int port  = atoi( str + i + 1 );
+  if( ( port > 0 && port < 1024 ) || port > (int)USHORT_MAX ) {
+    FD_LOG_ERR( ( "invalid port number" ) );
+    return NULL;
+  }
+  res->port = htons( (ushort)port );
+
+  return res;
+}
+
+FD_FN_CONST static inline ulong
+scratch_align( void ) {
+  return 4096UL;
```

Comment:
Does this need to be 4096? I didn't look at fd_repair_align, but it just should be max of the align of things under scratch_footprint